### ISSUE TITLE
scaling/weird results

### ISSUE THREAD

**Issue by zacman2400** [zaccharieramzi/tfkbnufft#36]
I have been doing some further tests, and I notice the scaling is maybe off or (more likely) I am implementing something wrong? I took radial k-space data, transformed it to image space, then transformed it back to k-space. The maximum of the absolute value of the k-space guess compared to my original k-space data is 2 orders of magnitude off. I also am having trouble with simple gradient descent experiments where I fail to get convergence. This is with norm set to ortho, results are worse with norm turned off.  Any idea what might be going wrong?

**Comment by chaithyagr** [comment]
Hello @zacman2400 ,

We need to note that the adjoint operator isn't an inverse operator in case of NUFFT.
When you take the radial k-space back to image space, the image is badly skewed due to lack of density compensation in Fourier space. Due to this, the forward nufft results in a k-space, usually in different scaling.

What I would recommend you to do is to please use density compensation. The example for this most be in the notebook...

**Comment by zaccharieramzi** [comment]
Hi @zacman2400 , I totally second what @chaithyagr just said.

This behavior is unfortunately expected. You can take a look at how we handled it in a [recent work](https://arxiv.org/abs/2101.01570) that we are presenting this week at ISBI.
An extended version of this work was also recently submitted, don't hesitate to ask me for it.

**Comment by zacman2400** [comment]
I should note I am using density compensation, sorry I should have mentioned that. Any recommendations when this is happening even with density compensation? Could it be the way my density compensation is scaled?

**Comment by zaccharieramzi** [comment]
Ah ok!
We have recently implemented a new version of the density-compensation following the scheme from Pipe et al. rather than what was implemented in `torchkbnufft` originally. See the patch here https://github.com/zaccharieramzi/tfkbnufft/pull/26

You can also check the example use in the [example notebook](https://github.com/zaccharieramzi/tfkbnufft/blob/master/notebooks/basic_examples_tf.ipynb). See how we have a different reconstruction using the Pipe et al. scheme.

If this doesn't fix your problems, could you share a reproducible example, preferably in a colab notebook?

**Comment by zacman2400** [comment]
Thank you for the suggestions...I have convinced myself that this isn't really a problem with NUFFTs..but just a difficulty of using them in the context of NNs. Thanks for the help!

**Comment by zaccharieramzi** [comment]
I am sorry we couldn't be of more help.

I am still very surprised by the orders of magnitude you mentioned which shouldn't occur. 

My experience with training NNs with the NUFFT has proven more succesful so if you have some time to write down a reproducible example don't hesitate to share it.
